usd_qtpy/render_util/turntable.py

```diff
@@ -0,0 +1,374 @@
+
+    # center geometry.
+    if subject_zup:
+        subject_center_translate = Gf.Vec3d(
+                                            -subject_centroid[0] / min_sizediff,
+                                            subject_centroid[2] / min_sizediff,
+                                            -subject_bounds_min[1] / min_sizediff
+                                            )
+    else:
+        subject_center_translate = Gf.Vec3d(
+                                            -subject_centroid[0] / min_sizediff,
+                                            -subject_bounds_min[1] / min_sizediff,
+                                            -subject_centroid[2] / min_sizediff
+                                            ) 
+    
+    subject_ref_xformable.AddTranslateOp(
+                          UsdGeom.XformOp.PrecisionDouble,"center_centroid")\
+                         .Set(subject_center_translate)
+    
+    # turn off the lights if GL
+    if renderer == "GL":
+        lights_prim = ttable_stage.GetPrimAtPath("/turntable_reference/scene/lights")
+        if lights_prim.IsValid():
+            lights_prim.GetAttribute("visibility").Set("invisible",0)
+
+    realstage_filename = R"./temp/turntable_assembly.usd"
+    realstage_filename = os.path.abspath(realstage_filename)
+
+    ttable_stage.Export(realstage_filename)
+
+    # frame range 1-100 in standard file
+    # get_file_timerange_as_string should be preferred, but it doesn't work atm.
+    frames_string = get_turntable_frames_string(length,frame_start,repeats)
+    render_path = os.path.join(export_path, "turntablefile_###.png")
+    render_path = os.path.abspath(render_path)
+
+    print("Rendering",frames_string,render_path)
+
+    realstage = Usd.Stage.Open(realstage_filename)
+
+    turntable_camera = next(playblast.iter_stage_cameras(realstage),None)
+    turntable_camera = UsdGeom.Camera(turntable_camera)
+    turntable_camera = framing_camera.camera_conform_sensor_to_aspect(
+        turntable_camera,
+        width,
+        height
+    )
+
+    playblast.render_playblast(realstage,
+                               render_path,
+                               frames=frames_string,
+                               width=1920,
+                               camera=turntable_camera,
+                               renderer=renderer,
+                               qt_report_instance=qt_report_instance)
+    
+    # explicitly free scene to make composite file available for deletion
+    del realstage
+
+    os.remove(subject_filename)
+    os.remove(realstage_filename)
+
+
+def file_is_zup(path: str) -> bool:
+    stage = Usd.Stage.CreateInMemory(path)
+    return framing_camera.get_stage_up(stage) == "Z"
+
+
+def get_file_timerange_as_string(path: str) -> str:
+    """
+    Attempt to get timerange from a USD file.
+
+    DOES NOT APPEAR TO WORK, EVEN WITH CORRECT METADATA.
+    """
+    stage = Usd.Stage.CreateInMemory(path)
```

Comment:
This is not _loading_ the file. It's creating an empty stage in memory that _will_ save to the given path when saved.

As such, it makes sense there's never authored time code range. You might be looking for [`Usd.Stage.Open()`](https://openusd.org/dev/api/class_usd_stage.html#abd74eb9ec70572c2dadaf4cd07144380)?

Reply:
I thought I already tried that but I could be wrong.

Either way, I've found that even with explicitly setting the layer attribute in e.g. Houdini,
the authored time code range still comes out wrong, however, that was on 19.5, I've yet to check on 20.

Reply:
Could you provide an example minimal usda file?

Reply:
I mostly checked with the turntable, but sure, I can provide you with a minimal usda file either this afternoon or tomorrow. I still have HKU to attend to today.